core/src/main/java/ai/timefold/solver/core/api/move/MutableSolutionState.java

```diff
@@ -0,0 +1,80 @@
+package ai.timefold.solver.core.api.move;
+
+import ai.timefold.solver.core.api.domain.metamodel.PlanningListVariableMetaModel;
+import ai.timefold.solver.core.api.domain.metamodel.PlanningVariableMetaModel;
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.domain.variable.PlanningListVariable;
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+import ai.timefold.solver.core.api.score.director.ScoreDirector;
+
+/**
+ * Contains all reading and mutating methods available to a {@link Move}
+ * in order to change the state of a {@link PlanningSolution planning solution}.
+ * <p>
+ * <strong>This package and all of its contents are part of the Move Streams API,
+ * which is under development and is only offered as a preview feature.</strong>
+ * There are no guarantees for backward compatibility;
+ * any class, method or field may change or be removed without prior notice,
+ * although we will strive to avoid this as much as possible.
+ * <p>
+ * We encourage you to try the API and give us feedback on your experience with it,
+ * before we finalize the API.
+ * Please direct your feedback to
+ * <a href="https://github.com/TimefoldAI/timefold-solver/discussions">Timefold Solver Github</a>.
+ * 
+ * @param <Solution_>
+ */
+public interface MutableSolutionState<Solution_> extends SolutionState<Solution_> {
```

Comment:
What about the unassign operation?

Reply:
This class technically doesn't need even the assign operation. :-) They are unused, prepared for the next phase. I'll leave this there, the other operations will show up in other PRs.